eng/SourceBuild.props

```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <GitHubRepositoryName>runtime</GitHubRepositoryName>
+    <NoTargetFrameworkFiltering>true</NoTargetFrameworkFiltering>
```

Comment:
Please move this here: https://github.com/dotnet/runtime/blob/2d9cb2d33f6f89f0a1a01782f2e474051bb2894a/Directory.Build.props#L293

Otherwise target framework filtering could happen outside of the "-sb" build when passing `/p:DotNetBuildFromSource=true` in.

Reply:
I thought that `ArcadeBuildFromSource` was basically the default at this point and that source build was not generally functional without it. That switch would bring in this file. @dotnet/source-build-internal?

Reply:
Linux partners actively use the `DotNetBuildFromSource` in their CI, cc @tmds

Reply:
Those are usually used as a combo though, I think? Otherwise SourceBuild.props wouldn't be the place to put any source build specific properties.

Reply:
Nope, they use the source build switch without depending on Arcade powered source build. See https://github.com/dotnet/runtime/pull/81480 for more details